### ISSUE TITLE
EdmChangedObjectCollection returns response with control annotations without "@odata" prefix regardless of version

### ISSUE THREAD

**Issue by Xriuk** [OData/AspNetCoreOData#1168]
**Assemblies affected**
ASP.NET Core OData 8.2.3

**Describe the bug**
When returning a delta payload via `EdmChangedObjectCollection`, the returned JSON response contains control annotations with short names (eg: "@context" instead of "@odata.context"), regardless of OData response version header (4.0 must return long ones, while 4.01 could return short ones).
Other results return like a regular list of entities produce correctly long annotations like "@odata.context".

**Reproduce steps**
Inside any controller, return a `EdmChangedObjectCollection` instance with few entries. The response should contain `OData-Version 4.0` header with "@context" annotation.

**Expected behavior**
If the response header is `OData-Version 4.0` the annotations should be long like "@odata.context".

**Comment by julealgon** [comment]
I didn't even know you could use this type directly? Sorry for the off-topic question, but is it documented anywhere? What are the use cases for using it?

You mentioned "returning a delta payload" @Xriuk , what kind of endpoint would do that? I was always under the impression that you only took deltas as input, but when returning you'd always return the full entity with the delta applied and let the user add projections if he needs.

**Comment by Xriuk** [comment]
@julealgon the type is not marked or documented as internal. As of OData protocol, if I'm not mistaken, the delta payload is supposed to be returned for delta resources (change tracking, from 4.0), and as an input and returned value for a delta patch of multiple entities to an entity set (from 4.01), like this one:
https://learn.microsoft.com/en-us/odata/webapi-8/fundamentals/entityset-routing?tabs=net60%2Cvisual-studio#patching-a-collection-of-entities
http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_UpdateaCollectionofEntities

**Comment by julealgon** [comment]
> @julealgon the type is not marked or documented as internal. As of OData protocol, if I'm not mistaken, the delta payload is supposed to be returned for delta resources (change tracking, from 4.0), and as an input and returned value for a delta patch of multiple entities to an entity set (from 4.01), like this one: https://learn.microsoft.com/en-us/odata/webapi-8/fundamentals/entityset-routing?tabs=net60%2Cvisual-studio#patching-a-collection-of-entities http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_UpdateaCollectionofEntities
>
>Currently I'm using it like this, the type mentioned here is different a bit, but it works in a similar way:
https://learn.microsoft.com/en-us/odata/webapi/DeltaFeed_support
And on JSON output it works, but not completely

Thank you so much for enlightening me about this! You are definitely right, this is mentioned explicitly in the spec:
> The response, if requested, is a delta payload, in the same structure and order as the request payload, representing the applied changes.

I'm seeing some interesting inconsistencies there though, mainly this:
> The body of the request MUST be a [delta payload](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_DeltaPayloads), **and the resource path of the collection MUST NOT contain type cast or filter segments**, and MUST NOT contain any system query options that affect the shape of the result.

Emphasis mine to contrast with the examples in that Microsoft Learn article:

![image](https://github.com/OData/AspNetCoreOData/assets/461579/79a3ddd8-b86f-4145-a2d6-effb4a255e1c)

Doesn't this request violate the bolded rule above? Isn't this a typecast in the PATCH request?
```
PATCH http://localhost:5000/odata/Shapes/EntitySetRouting.Models.Rectangle
```

**Comment by Xriuk** [comment]
@julealgon Yeah, don't know about that one, imho it makes sense to allow the cast since you either specify the type of the derived entities you are patching in the url with the type segment or in the body for each element in the payload with `@odata.type` annotation. The protocol might be at wrong, or not...

**Comment by xuzhg** [comment]
Currently, the Delta response payload follows up OData Version 4.01 format. It's hard-coded now at: https://github.com/OData/AspNetCoreOData/blob/main/src/Microsoft.AspNetCore.OData/Formatter/ODataOutputFormatterHelper.cs#L92-L94

This doesn't make sense to me also.
